hive/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -33,7 +34,7 @@
   }
 
   HiveClientPool(int poolSize, Configuration conf) {
-    super(poolSize, TException.class);
+    super(poolSize, TTransportException.class);
```

Comment:
@aokolnychyi, this was incorrect and causing the clients to reconnect on every `TException`, including exceptions thrown when tables already exist or do not exist (in the existence check). You will probably want to use this change to avoid over-reconnecting.

In addition, I think that it is correct to close the client before reconnecting because with this problem I was still seeing connection exhaustion in the test metastore. Closing before reconnect fixed the problem and using `TTransportException` also fixed the problem independently.

Reply:
+1, thanks! We will pull this in.